onnxoptimizer/passes/eliminate_nop_add_zero.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+// before: B = add(A, 0) or B = add(0, A)
+// after: B = A
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateNopAddZero final : public PredicateBasedPass {
```

Comment:
it's better to fuse the four passes into one `EliminateOpWithUnit`, which 
1. eliminates the operator when it represents an operation of monoid and one of its operands is the identity element, for example, multiplication with 1, addition with 0, pow with 1, max with std::numeric_limits<T>::max/infinity or min with std::numeric_limits<T>::min/-infinity. All these elimination can be implemented once by combining two functions  `void EliminateMonoidOpWithUnit(Kind op_kind)` and `T identityElement(Kind op_kind)`
2. eliminates the operator when it represents an operation that can be "transformed" to case 1, for example, subtraction of 0 and division of 1